src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+public abstract class Figure implements Draw, AreaCalculation {
+    protected String color1;
+    private Color color;
+
+    public Figure(){
+
+    }
+
+    public Figure(String color1) {
+        this.color1 = ColorSupplier.getRandomColor();
+    }

```

Comment:
```suggestion
```

random color must be passed from the outside, and not received inside the shape class